### ISSUE TITLE
Publish to npm

### ISSUE THREAD

**Issue by thisismydesign** [c-hive/team-contribution-calendar#27]
Depends on #25, #24, #18

Publish package to npm.

Acceptance criteria: works on `c-hive.dev` from npm. Follow semantic versioning. Start with `v0.1.0`. If AC is successful bump version to stable `v1.0.0`.

**Comment by thisismydesign** [comment]
Also add usage instructions to the readme via npm please.

**Comment by thisismydesign** [comment]
Use the following branch and tag setup for releases: https://github.com/c-hive/guides/blob/master/universal/lib.md#branches-and-tags

This means we might have to rethink how/when CDN content is generated. Is it possible to specify a branch for the script tag?

**Comment by gomorizsolt** [comment]
Sorry, didn't see your question here.

`jsDelivr` supports a bunch of different ways of defining the CDN. Take a quick glance at the examples on the [**official page**](https://www.jsdelivr.com/). It says that we can use:
- releases,
- branches
- or commits.

**Comment by thisismydesign** [comment]
TODO:
- [ ] update status and support chapter in readme
- [ ] update `You are viewing the README of the development version. There are no releases yet.`, in the same commit increase version
- [ ] use CDN from release branch and via version number
- [ ] test CDN via a v0.2.0 release

**Comment by thisismydesign** [comment]
Release flow:
- develop in PRs, merge to master as usual
- [master branch] when new release is ready, update readme's `Version information` part, update cdn link to upcoming release version (non existing yet), bump version
- create PR on GH from master to release (base: release, compare: master)
- squash merge PR
- checkout release branch locally
- tag squashed commit with version, e.g. `v0.3.0`
- travis CI script should pick up the new tag and release automatically

**Comment by thisismydesign** [comment]
OK after some reading I realized we can do whitelisting and it's the better approach.

https://docs.npmjs.com/misc/developers#keeping-files-out-of-your-package

The following paths and files are never ignored, so adding them to .npmignore is pointless:

package.json
README (and its variants)
CHANGELOG (and its variants)
LICENSE / LICENCE
If, given the structure of your project, you find .npmignore to be a maintenance headache, you might instead try populating the files property of package.json, which is an array of file or directory names that should be included in your package. Sometimes a whitelist is easier to manage than a blacklist.

> npm supports whitelisting though, just add a files attribute to package.json with everything you intend to add to the project. Now only the files that are specified in files will be included in the project and your dotfiles will be ignored. If you want to add a dotfile or a test directory (which would be odd), you’ll need to be explicit about it. npm will include the README, package.json, and a couple other files always so you don’t need to specify them.

See also: https://medium.com/@jdxcode/for-the-love-of-god-dont-use-npmignore-f93c08909d8d

**Comment by thisismydesign** [comment]
Here's how to properly initialize the release branch:

```
git rev-list --max-parents=0 master // => first commit sha1
git checkout <sha1> // => replace <sha1>
git checkout -b release
git merge --squash master
git add *
git commit --amend -m "v0.1.0"
```

Please remove it an initialize it like this during the next release.

**Comment by thisismydesign** [comment]
This doesn't work as the above results in `fatal: refusing to merge unrelated histories` the next time you try to `git merge master` on `release`. Perhpas we need a common initial commit in the repo first.

**Comment by gomorizsolt** [comment]
@thisismydesign What's up with this issue? It's already resolved by #53, isn't it?

**Comment by thisismydesign** [comment]
Yes :)
